las/client.py

```diff
@@ -89,303 +90,871 @@ def _create_signing_headers(self, path: str):
+        :return: User response from REST API
+        :rtype: dict
+
+        :raises: :py:class:`~las.InvalidCredentialsException`, :py:class:`~las.TooManyRequestsException`,\
+ :py:class:`~las.LimitExceededException`, :py:class:`requests.exception.RequestException`
+        """
+        return self._make_request(requests.delete, f'/users/{user_id}')
+
+    def create_workflow(self, specification: dict, name: str, description: Optional[str] = None,
+                        error_config: Optional[dict] = None) -> dict:
+        """Creates a new workflow, calls the POST /workflows endpoint.
+
+        >>> from las.client import BaseClient
+        >>> from pathlib import Path
+        >>> client = BaseClient()
+        >>> specification = {'language': 'ASL', 'version': '1.0.0', 'definition': {...}}
+        >>> error_config = {'email': '<error-recipient>'}
+        >>> client.create_workflow(specification, '<name>', '<description>', error_config)
+
+        :param specification: Specification of the workflow
+        :type specification: dict
+        :param name: Name of the workflow
+        :type name: str
+        :param description: Description of the workflow
+        :type description: str
+        :param error_config: Configuration of error handler
+        :type error_config: dict
+        :return: Workflow response from REST API
+        :rtype: dict
+
+        :raises: :py:class:`~las.InvalidCredentialsException`, :py:class:`~las.TooManyRequestsException`,\
+ :py:class:`~las.LimitExceededException`, :py:class:`requests.exception.RequestException`
+        """
+        body = {
+            'specification': specification,
+            'name': name,
+            'description': description,
+            'errorConfig': error_config,
+        }
+        return self._make_request(requests.post, '/workflows', body=dictstrip(body))
+
+    def list_workflows(self, max_results: Optional[int] = None, next_token: Optional[str] = None) -> dict:
+        """List workflows, calls the GET /workflows endpoint.
+
+        >>> from las.client import BaseClient
+        >>> client = BaseClient()
+        >>> client.list_workflows()
+
+        :param max_results: Maximum number of results to be returned
+        :type max_results: int
+        :param next_token: A unique token for each page, use the returned token to retrieve the next page.
+        :type next_token: str
+        :return: Workflows response from REST API
+        :rtype: dict
+
+        :raises: :py:class:`~las.InvalidCredentialsException`, :py:class:`~las.TooManyRequestsException`,\
+ :py:class:`~las.LimitExceededException`, :py:class:`requests.exception.RequestException`
+        """
+        params = {
+            'maxResults': max_results,
+            'nextToken': next_token,
+        }
+        return self._make_request(requests.get, '/workflows', params=params)
+
+    def update_workflow(self, workflow_id: str, *, name: Optional[str], description: Optional[str] = None) -> dict:
```

Comment:
Using '*' syntax here but not in most of the functions? I am in favor of using '*' for all optional arguments